conans/client/toolchain/cmake.py

```diff
@@ -0,0 +1,293 @@
+# coding=utf-8
+
+import os
+import textwrap
+from collections import OrderedDict, defaultdict
+
+from jinja2 import Template
+
+from conans.client.build.cmake_flags import get_generator, get_generator_platform, \
+    CMakeDefinitionsBuilder, get_toolset
+from conans.client.generators.cmake_common import CMakeCommonMacros
+
+
+# https://stackoverflow.com/questions/30503631/cmake-in-which-order-are-files-parsed-cache-toolchain-etc
+# https://cmake.org/cmake/help/v3.6/manual/cmake-toolchains.7.html
+# https://github.com/microsoft/vcpkg/tree/master/scripts/buildsystems
+
+
+class Definitions(OrderedDict):
+    _configuration_types = None
+
+    def __init__(self):
+        super(Definitions, self).__init__()
+        self._configuration_types = {}
+
+    def __getattribute__(self, config):
+        try:
+            return super(Definitions, self).__getattribute__(config)
+        except AttributeError:
+            return self._configuration_types.setdefault(config, dict())
+
+    @property
+    def configuration_types(self):
+        # Reverse index for the configuration_types variables
+        ret = defaultdict(list)
+        for conf, definitions in self._configuration_types.items():
+            for k, v in definitions.items():
+                ret[k].append((conf, v))
+        return ret
+
+
+class CMakeToolchain(object):
+    filename = "conan_toolchain.cmake"
+
+    _conan_set_libcxx = textwrap.dedent("""
```

Comment:
Hard to see the difference 😄  Here it uses `CONAN_CXX_FLAGS` and the one in `cmake_common` uses `CMAKE_CXX_FLAGS`. In the toolchain the objective is to populate `CMAKE_CXX_FLAGS_INIT` without touching `CMAKE_CXX_FLAGS`. Probably this can be refactored in the `cmake` generator. I'll have a look at it.